server/src/main/java/org/opensearch/index/translog/transfer/TranslogTransferMetadata.java

```diff
@@ -100,7 +100,7 @@ public String getFileName() {
                 RemoteStoreUtils.invertLong(primaryTerm),
                 RemoteStoreUtils.invertLong(generation),
                 RemoteStoreUtils.invertLong(createdAt),
-                nodeId,
+                String.valueOf(Objects.hash(nodeId)),
```

Comment:
Would it be possible to add a small test around generating 100-1000+ nodeIds and then ensuring that they do not collide.

Reply:
It is a uuid, so the hashes shouldn't collide. Since the generation of node id can change, this test is not protecting us from future collissions.